internal/tiltfile/tiltextension/plugin.go

```diff
@@ -129,6 +131,75 @@ func (e *Plugin) LocalPath(t *starlark.Thread, arg string) (localPath string, er
 // Otherwise, infers an extension object that points to the default repo.
 func (e *Plugin) ensureExtension(t *starlark.Thread, objSet apiset.ObjectSet, moduleName string) *v1alpha1.Extension {
 	extName := apis.SanitizeName(moduleName)
+
+	extSet := objSet.GetOrCreateTypedSet(&v1alpha1.Extension{})
+	if existing, exists := extSet[extName]; exists {
+		ext := existing.(*v1alpha1.Extension)
+		metav1.SetMetaDataAnnotation(&ext.ObjectMeta, v1alpha1.AnnotationManagedBy, "tiltfile.loader")
+		return ext
+	}
+
+	repoSet := objSet.GetOrCreateTypedSet(&v1alpha1.ExtensionRepo{})
+
+	return e.registerExtension(t, extSet, repoSet, extName, moduleName)
+}
+
+// In cases where an extension is not already registered, this function will search for an extension
+// repo that can satisfy the requested extension, with a fallback to an extension in the default
+// repository.
+func (e *Plugin) registerExtension(t *starlark.Thread, extSet, repoSet apiset.TypedObjectSet, extName, moduleName string) *v1alpha1.Extension {
+	loadHost, extPath, tryRegister := strings.Cut(moduleName, "/")
+
+	// Safety fallback (in case this is called without already previously calling ensureExtension)
+	existing := extSet[extName]
+	if existing != nil {
+		return existing.(*v1alpha1.Extension)
+	}
```

Comment:
This seems a bit off. In a fuller implementation this would also set the `tiltfile.loader` annotation but then it's effectively incorporating `ensureExtension`. Maybe there's a better way to break up `ensureExtension`. I'll think on it.